repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner.py

```diff
@@ -339,13 +385,34 @@ def parse_action(action_id):
     return Action(action_id)
 
 
+def parse_modulestream(package):
+    modulestreams = package.get('modulestreams')
+    # in an older version of PES data, the field for module stream can be absent
+    if not modulestreams:
+        return [None]
+    return ((modulestream['name'], modulestream['stream']) if modulestream else None for modulestream in modulestreams)
+
+
+def parse_package(package):
+    return Package(package['name'], package['repository'], parse_modulestream(package))
+
+
 def parse_packageset(packageset):
     """
     Get "input" or "output" packages and their repositories from each PES event.
 
-    :return: A dictionary with packages in format {<pkg_name>: <repository>}
+    :return: set of Package tuples
     """
-    return {p['name']: p['repository'] for p in packageset.get('package', [])}
+    return {parse_package(p) for p in packageset.get('package', packageset.get('packages', []))}
```

Comment:
This raises a `TypeError: unhashable type: 'list'`

This works:
```suggestion
    return set(parse_package(p) for p in packageset.get('package', packageset.get('packages', [])))
```